dist/ExtUtils-ParseXS/lib/perlxs.pod

```diff
@@ -1,2065 +1,4803 @@
+temps stack is reset at the start of each statement, back to some
+particular level. Each SV above this level has its reference count
+decremented. Putting an SV on the temps stack is referred to as
+I<mortalising> it. It is common to create a new SV and mortalise it at the
+same time: here are some examples:
+
+    SV *sv_99  = sv_2mortal(newSViv(99));
+    SV *sv_abc = newSVpvn_flags("abc", 3, SVs_TEMP);
+
+Many OPs have an SV attached to them called a I<PADTMP>. This SV has a
+long lifetime which is the same as the sub which the OP is a part of
+(typically it is created when a named sub is compiled and freed when that
+sub is deleted, often at the end of the execution of the program),
+and usually has a reference count of one. It is used by many OPs to avoid
+having to create (and later free) a temporary SV to return a value. For
+example the C<ADD> op in C<$a + $b> typically extracts the integer values
+of its two arguments, calculates its sum, sets its C<PADTMP> to that value
+and pushes it onto the stack. The C<OP_ENTERSUB> which typically invokes
+an XSUB usually has a C<PADTMP> attached to it, and when returning a
+value, the XSUB's boilerplate code generated by XS will usually try to use
+it to return the value, rather than creating a fresh mortal SV on each
+call.
+
+Note that there is a I<highly experimental> perl interpreter build option,
+C<PERL_RC_STACK>, under which the argument stack I<is> reference counted,
+but that is currently beyond the scope of this document.
+
+=head3 The IV, NV etc types
+
+An C<IV> (Integer Value) is a typedef in the perl interpreter's header
+files that maps to a C integer. The exact integer type and size will
+depend on the build configuration of the interpreter. It is guaranteed to
+be large enough to hold a pointer. A C<UV> is the same but unsigned. An
+C<NV> (numeric value) is a floating-point value; usually a C<double>.
+These types are used widely within the perl interpreter.
+
+A PV (pointer value) "type" is often used informally within documentation
+and within the names of structure fields etc to refer to a string pointer
+(C<char*>), but it is not actually a declared type. Similarly, RV
+(reference value) is informally a pointer to another SV.
+
+There are also C<SSize_t> and C<Size_t>, which are large enough to hold
+signed and unsigned integer values representing the number of items in a C
+array. C<STRLEN> is used specifically for variables which store the number
+of characters in a string (it is typically just an alias for C<Size_t>).
+
+=head3 The SV scalar value structure
+
+(Again there is a lot of detail in this section that you may not need to
+know when just creating simple XSUBs, but is useful background for
+debugging.)
+
+As mentioned above, almost all runtime data within the perl interpreter is
+stored in an SV (scalar value) structure. The head of an SV structure
+consists of three or four words: a reference count; a type and flags; a
```

Comment:
I think "words" is possibly not a word [pardon the phrasing] that would make sense to folks outside of a machine-level programming background. I'd say "fields", just relating to the members of the C struct.